components/brave_wallet/browser/solana_tx_manager_unittest.cc

```diff
@@ -1423,4 +1427,83 @@ TEST_F(SolanaTxManagerUnitTest, ProcessSolanaHardwareSignature) {
+
+  // No rebroadcast when first sending out the tx.
+  ApproveTransaction(meta_id1);
+  task_environment_.RunUntilIdle();
+  EXPECT_EQ(send_transaction_calls_, 1u);
+
+  // Rebradcast should happen once for tx with null signature status.
+  send_transaction_calls_ = 0;
+  SetInterceptor(latest_blockhash2_, last_valid_block_height2_, "", "", true,
+                 last_valid_block_height1_, "[null]");
+  task_environment_.FastForwardBy(
+      base::Seconds(kSolanaBlockTrackerTimeInSeconds));
+  task_environment_.RunUntilIdle();
+  EXPECT_EQ(send_transaction_calls_, 1u);
+
+  // Rebroadcast should happen once when tx is not confirmed and latest
+  // blockhash is updated which triggers UpdatePendingTransaction.
+  send_transaction_calls_ = 0;
+  SetInterceptor(latest_blockhash3_, last_valid_block_height3_, "", "", true,
+                 last_valid_block_height1_,
+                 R"([{"slot": 100, "confirmations": 100, "err": null,
+                      "confirmationStatus": "processed"}])");
+
+  task_environment_.FastForwardBy(
+      base::Seconds(kSolanaBlockTrackerTimeInSeconds));
+  task_environment_.RunUntilIdle();
+  EXPECT_EQ(send_transaction_calls_, 1u);
+
+  // No rebroadcast when blockhash is expired.
+  send_transaction_calls_ = 0;
+  SetInterceptor(latest_blockhash1_, last_valid_block_height1_, "", "", true,
+                 last_valid_block_height2_,
+                 R"([{"slot": 100, "confirmations": 100, "err": null,
+                      "confirmationStatus": "processed"}])");
+  task_environment_.FastForwardBy(
+      base::Seconds(kSolanaBlockTrackerTimeInSeconds));
+  task_environment_.RunUntilIdle();
+  EXPECT_EQ(send_transaction_calls_, 0u);
+
+  SetInterceptor(latest_blockhash2_, last_valid_block_height2_, "", "", true,
+                 last_valid_block_height2_, "[null, null]");
+  data->send_options = mojom::SolanaSendTransactionOptions::New(
+      mojom::OptionalMaxRetries::New(1u), std::nullopt, nullptr);
+  std::string meta_id2;
+  AddUnapprovedTransaction(mojom::kSolanaMainnet, data.Clone(), sol_account(),
+                           &meta_id2);
+  ASSERT_FALSE(meta_id2.empty());
+
+  // Avoid cached blockhash being used when approve.
+  task_environment_.FastForwardBy(
+      base::Seconds(kSolanaBlockTrackerTimeInSeconds));
+  task_environment_.RunUntilIdle();
+
+  // No rebroadcast when first sending out the tx with customized MaxRetries.
+  send_transaction_calls_ = 0;
+  ApproveTransaction(meta_id2);
+  task_environment_.RunUntilIdle();
+  EXPECT_EQ(send_transaction_calls_, 1u);
+
+  // Rebroadcast when tx signature status is null with customized MaxRetries.
+  SetInterceptor(latest_blockhash3_, last_valid_block_height3_, "", "", true,
+                 last_valid_block_height2_, "[null]");
+  send_transaction_calls_ = 0;
+  task_environment_.FastForwardBy(
+      base::Seconds(kSolanaBlockTrackerTimeInSeconds));
+  task_environment_.RunUntilIdle();
+  EXPECT_EQ(send_transaction_calls_, 1u);
+}
```

Comment:
We miss one case for dropping tx after `kSafeDropThresholdInMinutes`

Reply:
Fixed in https://github.com/brave/brave-core/pull/23329/commits/7cc18ad5a5f0ef1ca890d6eb24f5d3da6a50a255